### PR TITLE
[WIP] Update accessibility.md

### PR DESCRIPTION
**Pull request by atoedt**
changes on cheat sheet: 
* changes are supposed to reflect discussion in SIG accessibility on process (i.e. title) 
* changes supposed to reduce number of things to check to the utmost minimum (leaving out second rank issues) 
* changes supposed to reduce ambivalences and improve decidability

This is still work in progress and not finished yet, sorry.

### CODE REVIEW

**Comment by mjansenDatabay** [comment]
@atoedt See #2860

File: docs/development/accessibility.md [hunk 1]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
-3.	Users must be able to skip ‘repeated content of second rank’ i.e. Secondary Panels (2.4.1). Use aria landmark role ‘complementary’ to make elements skippable. 
-4.	Make the DOM order matche the visual order (1.3.2). When the source order matches the visual order, everyone will read the content and interact with it in the same (correct) order. If this will not work use tabindex 0. 
-5.	Make sure that repeated navigational components occur in the same place and same relative order each time they appear. (3.2.3) 
-## Headings and Labels and Language
-6.	Our headings and labels have to be clear and descriptive (2.4.6). 
-7.	Provide clear and descriptive labels for any field that requires input. AND one of the following (3.3.2)
+Check how your project complies to our semantic structures: 
+
+8.	Write “well formed” HTML, employ HTML according to the specification with complete start and end tags and compliant nesting. Assistive technology will crash on missing start or end tags. (4.1.1). Use [validator](https://validator.w3.org/) to verfiy, [this test case](https://testrail.ilias.de/index.php?/tests/view/44245) may be helpful.
```
Comment (atoedt) [inline docs/development/accessibility.md]: correct attributes,

Comment (atoedt) [inline docs/development/accessibility.md]: nur valides HTML, der restliche Text kann weg

**Comment by alex40724** [comment]
I marked this as WIP

**Comment by alex40724** [comment]
Please assign to JourFixe when ready and before merging.

**Comment by tfamula** [comment]
@atoedt Please have a look into #2903

File: docs/development/accessibility.md (hunk 1 above)
Comment (nicoroeser) [reply docs/development/accessibility.md]: In case you keep it: `verfiy` → `verify`.

File: docs/development/accessibility.md [hunk 2]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
+
+18.	Non-text elements like icons, images, graphs and the like must have alt-texts (1.1.1). 
+19.	Glyphs are done in spans which cannot bear alt-texts but carry a link-tag, which must have an aria-label.  
+20.	Use aria-label to provide an invisible label where a visible label cannot be used. (4.1.2)
+21.     Ensure labeling requirements are met. Use Wave or Lighthouse to verify. 
+
+
+
+
+
```
Comment (nicoroeser) [inline docs/development/accessibility.md]: Trailing whitespace in file.

File: docs/development/accessibility.md [hunk 3]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
-This list is prepared to inform developers what accessibility means for their daily work. This list was prepared by the SIG accessibility to their best knowledge. Still it may contain some errors. 
-Thus, we suggest to use compliance testing as a yardstick for quality: Compliance testing is more important than these phrases, we will adapt these phrases via the jour fixe if compliance testing indicates they are erroneous or lacking information.  
-We still hope the following list is helpful in everyday work.
+# Accessibility Checklist 
+This list is prepared to be worked thorugh by developers implementing specific projects. Results from external compliance testing may lead to adapting these phrases. Changes will be subjected to the jour fixe if compliance testing indicates they are erroneous or lacking information. We still hope the following list is helpful in everyday work, please let us know if this is not the case. 
+1. Tastaturnutzbarkeit
+2. Fokusverfolgung
+3. HTML Elemente Überschriftenstruktur
+4. Kontraste
+5. Alt-Texte
```
Comment (kergomard) [inline docs/development/accessibility.md]: Why is this in German?

File: docs/development/accessibility.md [hunk 4]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
+## Tabbing and Focus
+Please manually check your project for tabbing and focus quaility: 
 
+1.	All functionality must be operable by tabbing or number pad arrows. (2.1.1) 
+2.	Users can tab to any control, operate it, and tab away again. (2.1.2) 
+3.	Keyboard focus must be always visible. The element that has focus gets a corona. (2.4.7) 
+4.	Tabbing order must be sensible and predictable. The move of focus is following the meaning of the content and can be operated by tab. Moving focus should not jump around unpredictably. (2.4.3)
```
Comment (kergomard) [inline docs/development/accessibility.md]:
```suggestion
4.	Tabbing order must be sensible and predictable. The focus moves following the meaning of the content and can be operated by tab. Moving focus should not jump around unpredictably. (2.4.3)
```

File: docs/development/accessibility.md [hunk 5]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
-2.	We use aria landmarks and headlines to furnish ILIAS with semantic structures. Regions of the page and navigation are structured by aria landmarks. Content gets structured by headlines. These elements provide programmatically determinable semantic structures to convey information and relationships (1.3.1).  
-3.	Users must be able to skip ‘repeated content of second rank’ i.e. Secondary Panels (2.4.1). Use aria landmark role ‘complementary’ to make elements skippable. 
-4.	Make the DOM order matche the visual order (1.3.2). When the source order matches the visual order, everyone will read the content and interact with it in the same (correct) order. If this will not work use tabindex 0. 
-5.	Make sure that repeated navigational components occur in the same place and same relative order each time they appear. (3.2.3) 
-## Headings and Labels and Language
-6.	Our headings and labels have to be clear and descriptive (2.4.6). 
-7.	Provide clear and descriptive labels for any field that requires input. AND one of the following (3.3.2)
+Check how your project complies to our semantic structures:
```
Comment (kergomard) [inline docs/development/accessibility.md]:
```suggestion
Check how your project complies with our semantic structures: 
```

File: docs/development/accessibility.md [hunk 6]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
+9.	We use aria landmarks and headlines to furnish ILIAS with semantic structures. Regions of the page and navigation are structured by aria landmarks. Content gets structured by headlines. These elements provide programmatically determinable semantic structures to convey information and relationships (1.3.1).  @Thomas: Ich brauche hier eine Sprungstelle auf das md zu den Roles.  @Michael: Ich brauche hier eine Sprungstelle zu den Headlines. 
+10.	Make the DOM order match the visual order (1.3.2). The screen reader will the DOM aloud: When the source order matches the visual order, everyone will read the content and interact with it in the same (correct) order. Elements visually presented on the right hand-side should in the DOM be located after elements visually presented on the left-hand-side . Elements visually presented on the bottom should in the DOM be located after elements visually on the top.
```
Comment (kergomard) [inline docs/development/accessibility.md]:
```suggestion
10.	Make the DOM order match the visual order (1.3.2). The screen reader will read the DOM aloud: When the source order matches the visual order, everyone will read the content and interact with it in the same (correct) order. Elements visually presented on the right hand-side should in the DOM be located after elements visually presented on the left-hand-side . Elements visually presented on the bottom should in the DOM be located after elements visually presented on the top.
```

File: docs/development/accessibility.md [hunk 7]
```diff
@@ -776,76 +776,52 @@ Target group: developer
...
+11.	Make sure that repeated navigational components occur in the same place and same relative order each time they appear. (3.2.3) 
+## Headings and Labels and Language (Überschrift ist unpassend) 
+12.	Our headings and labels have to be clear and descriptive (2.4.6). This does not neccessarily concern developers and can be handled by language manintainer or editorial team.
```
Comment (kergomard) [inline docs/development/accessibility.md]:
```suggestion
12.	Our headings and labels have to be clear and descriptive (2.4.6). This does not necessarily concern developers and can be handled by language maintainer or  the editorial team. 
```

**Review by kergomard — COMMENTED** [review]
Just a few little editorial inputs, if this is kept as is. Thanks for all the work @tfamula .

**Comment by alex40724** [comment]
Thanks for the PR!

Some personal suggestions from my side:

- Tabbing and Focus, 3: "The element that has focus gets a corona." > "The element that has focus is visually highlighted." (this is less specific)
- Tabbing and Focus, 7: "Give users a submit-button to initiate change of context." This is misleading imo and might be controversial. I would remove this point for now. Submit buttons should initiate system changes (form submissions), Links should indicate "change of context" (switching to other ILIAS screens/views).
- Structure, 14: "Provide clear and descriptive labels for any field that requires input." I think it would be wise, if the Jour Fixe would decide on labels when a feature is scheduled for a release. Developers alone may struggle with this point. I would remove this from the list and discuss it separately.
- Alt-Texts and aria labels, 18: I would change "Non-text elements like icons, images, graphs and the like must have alt-texts" to "img tags must get an alt attribute. The attribute should be empty for decorative images."
- Alt-Texts and aria labels, 20: I just learned, that aria-labels are ignored by screen readers on several elements. I would remove this point for now, investigate on this and add it later again. It might be better to work with "aria-only" HTML text (only read by screen readers, hidden otherwise) in some cases imo.

And proper translations for the remaining German text should be added as well as the references to the other .md files.

**Comment by Amstutz** [comment]
@smeyer-ilias : Why has the PR been closed? Note that this is still on the list for the TB, however, we tackle one piece at the time. Currently we are still waiting to resolve the discussion around the accessibility process, see: https://github.com/ILIAS-eLearning/ILIAS/pull/3190

**Comment by smeyer-ilias** [comment]
@Amstutz : The related branch (atoedt-patch-1) was accidently pushed to the ILIAS repository. I deleted this branch but was not aware that is related to an open PR.

**Comment by Amstutz** [comment]
@smeyer-ilias : ahh, I see. makes sense. I did wonder, what this branch was for in the main repo. Thx for cleaning up. Maybe, we close this here, open a new one, and just link the two?

**Comment by klees** [comment]
Hi everyone,

I would like to point out a few problems that I see with this document (and hence this PR):

* With the [initial introduction of the guidelines](https://github.com/ILIAS-eLearning/ILIAS/pull/1944#issuecomment-534089769) promises were made (by @atoedt) that have not been realized yet:
   * a procedure how ILIAS can be made guideline-conformant
   * a training for developers
   * constantly accessible documents to allow new developers to get the needed know-how on how to create compliant code
   * funding for tickets created based on this guideline
* The idea to solve accessibility by "Guidelines" or "Checklists" is subkomplex. The WCAG itself is widely open for interpretation. In the original version of our guideline there has been a cheat sheet for developers that had given some of these interpretation. This will be transformed into a check list, that as of now is already incompatible with some of the issues that have been (rightfully) reported. Accepting this PR will move us even further in a direction, where we have contradictory requirements in guidelines, tickets and discussions, since this PR will lift the "cheat sheet" to a "check list".
* With the introduction of the PR I already pointed out that it will be hard to find or instruct developers to work on our UI. Hence the (unfulfilled) promise to provide additional documents and training aimed at developers. This documents remain to be created, the training remains to be carried out. An according request by the TB for offers to create some part of that documentation ("Call for Bids : Accessibility Tool Documentation", 29.10.2021) was unanswered. I currently can not see how I could, with good consciousness, let any of my developers work on UI components in the community anymore. The atmosphere regarding A11y is mostly toxic, requirements are contradictory and documentation is lacking. As pointed out before, this will get worse with this PR, since it would elevate the "cheat sheet" to a "check list". We should be roping in new developers (e.g. by providing a nice atmosphere to work in), but instead this will alienate existing developers (like myself!) even further.

The idea that a good accessibility process can be implement by the means of guidelines has been tested for 1 1/2 year now. From my perspective the consequences for our community have been devastating so far. We luckily made some progress regarding the accessibility of ILIAS, but a lot of time has also been spend for unproductive accusations, bickering and back-and-forth in various discussions. Imagine where we could have been if we used all that energy productively.

Hence I, as Richard from CaT, UI coordinator and contributor to the ILIAS UI, ask the JF to clearly and unambigiously state, that the process and not guidelines will need to be emphasized to get to an accessible ILIAS. I ask the JF to drop this work on the "Accessibilty by Guideline" idea for now (or surround it with an according set of ifs and thens that address my issues above) and instead (again) embrace the "Accessibility by Process" idea, as outlined in [the Accessibility Process](https://github.com/ILIAS-eLearning/ILIAS/blob/653d7c070d009f2d27ac43e8e3479e32aec4d6ba/docs/development/accessibility-process.md). Guidelines and checklists will play some part in that process, but cooperation (and not legislation) will be key to move towards an accessible system.

Best regards!

**Comment by Amstutz** [comment]
For me important, that we all agree, that all content bellow the title:

> # Accessibility Checklist

  is a **checklist** NOT a guideline. It is suposed to help to give developers some concrete handles on how to check their code towards WCAG.  The points listed in there may not apply in every given scenario (e.g. see 18 and 19 for icons, images and glyphs, which collide in certain scenarios such as bulky buttons, see: https://mantis.ilias.de/view.php?id=30308) and might be subject to change in the near future when we have more expertise in the community on the mater.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 03 MAY 2021 : Alexandra Tödt and Wolfgang Schmidt-Sielex want to continue working on this document. Therefore, we keep the PR open but remove TB label and assignees.

**Review by atoedt — COMMENTED** [review]
done

**Comment by Amstutz** [comment]
The TB discussed the PR today, since it showed up it the list of old PRs we monitored regularly. In my role as Head of the SIG accessibility I informed the TB, that this PR is currently in its final stages of discussion in the SIG before being discussed at the JF. If you have stakes in this, plz join the discussions in the SIG a11y.

**Comment by Amstutz** [comment]
We discussed the current version of the proposal today in the SIG, and recommend this to be merged with trunk.

**Comment by Amstutz** [comment]
Sorry, for this late change Proposal, but we are learning every day:

We (FH Luzern and Bern) had again our consulting meeting with Webkey. There have been some takeaways from that meeting concerning headlines. 

- We learned that there is no such strict requirement for having "no holes" in the structure. In contrary, it can make sense e.g. in the header to start with h2 to not have screenreaders always reading the same page title or similar. This "MUST" requirement should be reduced to "SHOULD". 
- Further we would drop "* There can be more than one H1 level per page. ", this contradicts "There MUST be exactly one single h1 level per page."

Since the SIG agreed on that being merged in the current state, we are fine with merging a proposing the above change later. If we all agree, we can also add this change before merging.

@amstutz and @swiniker

**Comment by matthiaskunkel** [comment]
Jour Fixe, 26 JUL 2021 : We highly appreciate this well elaborated document and accept it as a first version of our accessibility checklist for developers - incl. the two changes suggested by Timon recently. Nevertheless, this document will be constantly improved and revised and get new version in the future - according to our growth in knowledge for creating an accessible ILIAS. Please merge to trunk!

**Pull request merged**